src/originalBudgetTrackingApp/index.ts

```diff
@@ -104,48 +103,25 @@ export function calculateTransactionHash({
 }
 
 async function createTransactionsInExternalVendors(config: Config, companyIdToTransactions: Record<string, EnrichedTransaction[]>, startDate: Date) {
-  await ynab.init(config);
-  const activeVendors: any = [];
-  if (config.outputVendors.ynab?.active) {
-    activeVendors.push({
-      name: 'ynab',
-      createTransactionFunction: ynab.createTransactions,
-      options: config.outputVendors.ynab.options,
-    });
-  }
-  if (config.outputVendors.googleSheets?.active) {
-    activeVendors.push({
-      name: 'googleSheets',
-      createTransactionFunction: googleSheets.createTransactionsInGoogleSheets,
-      options: config.outputVendors.googleSheets.options,
-    });
-  }
   const executionResult = {};
   const allTransactions = _.flatten(Object.values(companyIdToTransactions));
 
-  if (!activeVendors.length) {
-    throw new Error('You need to set at least one output vendor to be active');
-  }
-
-  for (let j = 0; j < activeVendors.length; j++) {
-    const vendor = activeVendors[j];
-    const vendorConfig = config.outputVendors[vendor.name];
-    if (vendorConfig && vendorConfig.active) {
-      const vendorResult = await createTransactionsInVedor(vendor, allTransactions, startDate);
-      executionResult[vendor.name] = vendorResult;
+  for (let j = 0; j < outputVendors.length; j++) {
+    const outputVendor = outputVendors[j];
+    if (outputVendor.isActive(config)) {
+      if (outputVendor.init) {
+        await outputVendor.init(config);
+      }
+      console.log(`Start creating transactions in ${outputVendor.name}`);
+      const vendorResult = await outputVendor.exportTransactions(allTransactions, startDate, config);
+      console.log(`Finished creating transactions in ${outputVendor.name}`);
+      executionResult[outputVendor.name] = vendorResult;
     }
   }
-  return executionResult;
-}
-
-async function createTransactionsInVedor(vendor, transactions: EnrichedTransaction[], startDate: Date) {
-  console.log(`Start creating transactions in ${vendor.name}`);
-  const vendorResult = await vendor.createTransactionFunction(transactions, startDate, vendor.options);
-  if (vendorResult) {
-    console.log(`${vendor.name} result: `, vendorResult);
+  if (!Object.keys(executionResult).length) {
+    throw new Error('You need to set at least one output vendor to be active');
```

Comment:
I didn't understand this Error, and now more.
You already did the (none) output, so what is the error?
You just need to warn, with `console.warn` for example.

I know, there is no meaning of scraping without exporting, because we don't save any state, but still, something feels wired to me. Maybe because I'm running the process without output, for test the scraping, and **I know** there are no outputs, so why it's error?

Reply:
This error exists because when setting up the project you could easily forget to set any output vendor as active and then it would look as if all went well but actually nothing happened.

This type of check should be an indication in the ui instead of an error thrown. I am all for removing it, once we add an indication in the ui thats makes it clear that you have no active exporters defined.

Reply:
Achla, this is kind of what I mean.